includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,20 +2,44 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+    const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+    const STATUS_IMAGE_EXTRACTION_FAILED= 'failed';
 
-	static public function extract( $url ) {
+    static public function extract( $urls ) {
```

Comment:
> we need to change the name of the filter I modified so that any existing downstream hooks that rely on its original parameters/behavior do not get invoked when the filters for it are applied.

Yes, exactly that. Ideally we'd still be able to provide backwards compatibility here and the old filter continues to work as well. Maybe we can can run each image/dimension through the old filter after the fact?